### PR TITLE
Refactoring the mobile package

### PR DESCRIPTION
**Pull request by jessicafalk**
#### Description
AndroidClient and iOSClient don't exist anymore. There is now one MobileClient that can be used for either Android or iOS.
Also added new menu items to launch the apk either for a local or a cloud deployment. This allows us to not have to rebuild the worker everytime you want to switch between connecting to a cloud or a local deployment. 
Got  rid of the mobile screen we had. Now you just connect as soon as the game starts.

#### Tests
Ran it locally and in cloud for Android & iOS.

#### Documentation
coming soon 

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: workers/unity/Packages/com.improbable.gdk.mobile/Utility/DeviceInfo.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                        product.Equals("google_sdk"))
+                        ? MobileDeviceType.Virtual
+                        : MobileDeviceType.Physical;
+                }
+#elif UNITY_IOS
+                activeDeviceType = SystemInfo.deviceModel.Equals("x86_64")
+                    ? MobileDeviceType.Virtual
+                    : MobileDeviceType.Physical;
+#else
+                throw new PlatformNotSupportedException($"{nameof(DeviceInfo)} is only supported using the iOS platform.");
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.mobile/Utility/DeviceInfo.cs]:
```suggestion
                throw new PlatformNotSupportedException($"{nameof(DeviceInfo)} is only supported while using the iOS or Android platforms.");
```

File: workers/unity/Packages/com.improbable.gdk.mobile/Worker/DefaultMobileWorkerConnector.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+            if (string.IsNullOrEmpty(environment))
+            {
+                environment = PlayerPrefs.GetString(RuntimeConfigNames.Environment, string.Empty);
+            }
+            else
+            {
+                PlayerPrefs.SetString(RuntimeConfigNames.Environment, environment);
+            }
+
+            ShouldConnectLocally = environment == "local";
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.mobile/Worker/DefaultMobileWorkerConnector.cs]: We should probably store the possible values of `environment` as `const`s somewhere since we refer to raw strings in multiple files

File: workers/unity/Packages/com.improbable.gdk.mobile/Worker/DefaultMobileWorkerConnector.cs [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            InitializeClient();
+        }
+
+        protected virtual void InitializeClient()
+        {
+            var arguments = LaunchArguments.GetArguments();
+            var environment = CommandLineUtility.GetCommandLineValue(arguments, RuntimeConfigNames.Environment, string.Empty);
+            if (string.IsNullOrEmpty(environment))
+            {
+                environment = PlayerPrefs.GetString(RuntimeConfigNames.Environment, string.Empty);
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.mobile/Worker/DefaultMobileWorkerConnector.cs]: What if the player prefs isn't set?

File: workers/unity/Packages/com.improbable.gdk.mobile/Worker/DefaultMobileWorkerConnector.cs [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+                switch (DeviceInfo.ActiveDeviceType)
+                {
+                    case MobileDeviceType.Virtual:
+                        return DeviceInfo.AndroidEmulatorDefaultCallbackIp;
+                    case MobileDeviceType.Physical:
+                        return PlayerPrefs.GetString(HostIpPlayerPrefsKey);
+                }
+            }
+
+            if (!string.IsNullOrEmpty(IpAddress))
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.mobile/Worker/DefaultMobileWorkerConnector.cs]: We should document the order of precedence for these levels of "overrides"

File: workers/unity/Packages/com.improbable.gdk.mobile/Worker/DefaultMobileWorkerConnector.cs [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+            {
+                return hostIp;
+            }
+
+            if (Application.isMobilePlatform)
+            {
+                switch (DeviceInfo.ActiveDeviceType)
+                {
+                    case MobileDeviceType.Virtual:
+                        return DeviceInfo.AndroidEmulatorDefaultCallbackIp;
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.mobile/Worker/DefaultMobileWorkerConnector.cs]: What about iOS?

File: workers/unity/Packages/com.improbable.gdk.tools/GdkToolsConfigurationWindow.cs [hunk 6]
```diff
@@ -13,9 +13,10 @@ public class GdkToolsConfigurationWindow : EditorWindow
         internal const string SchemaStdLibDirLabel = "Standard library";
         internal const string CodegenOutputDirLabel = "Code generator output";
         internal const string SchemaSourceDirsLabel = "Schema sources";
+        internal const string MobileSectionLabel = "Mobile Settings";
         internal const string RuntimeIpLabel = "Local runtime IP";
         internal const string DevAuthTokenSectionLabel = "Dev Auth Token Settings";
-        internal const string DevAuthTokenDirLabel = "Directory to store token in";
+        internal const string DevAuthTokenDirLabel = "Dir to store token in";
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.tools/GdkToolsConfigurationWindow.cs]: Can you revert this change?

File: workers/unity/Packages/com.improbable.gdk.mobile/Worker/DefaultMobileWorkerConnector.cs (hunk 3 above)
Comment (jessicafalk) [reply workers/unity/Packages/com.improbable.gdk.mobile/Worker/DefaultMobileWorkerConnector.cs]: that's the bug I just encountered on iOS as we don't use command line arguments yet. fixing it right now

File: workers/unity/Packages/com.improbable.gdk.tools/GdkToolsConfigurationWindow.cs (hunk 6 above)
Comment (jessicafalk) [reply workers/unity/Packages/com.improbable.gdk.tools/GdkToolsConfigurationWindow.cs]: but then the text won't fit

File: CHANGELOG.md [hunk 7]
```diff
@@ -2,10 +2,14 @@
 
 ## Unreleased
 
+### Breaking Changes
+- Removed the `Improbable.Gdk.Mobile.Android` and `Improbable.Gdk.Mobile.iOS` packages. All functionality is now available inside the `Improbable.Gdk.Mobile` package.
```
[the remark this answers is not in the document]
Comment (jessicafalk) [reply CHANGELOG.md]: they were in the playground. So kinda?

File: .buildkite/premerge.steps.yaml [hunk 8]
```diff
@@ -52,7 +52,7 @@ steps:
       - logs/**/*
       - workers/unity/build/worker/**/*
     env:
-      WORKER_TYPE: "AndroidClient"
+      WORKER_TYPE: "MobileClient"
```
Comment (jamiebrynes7) [inline .buildkite/premerge.steps.yaml]: Can you change the label as well? 🙏

File: .buildkite/release-qa.steps.yaml [hunk 9]
```diff
@@ -38,7 +38,7 @@ steps:
       - logs/**/*
       - build/assembly/**/*
     env:
-      WORKER_TYPE: "AndroidClient"
+      WORKER_TYPE: "MobileClient"
```
Comment (jamiebrynes7) [inline .buildkite/release-qa.steps.yaml]: Change label here as well please 🙏

File: workers/unity/Packages/com.improbable.gdk.mobile/Utility/DeviceInfo.cs [hunk 10]
```diff
@@ -0,0 +1,54 @@
+using System;
+using UnityEngine;
+
+namespace Improbable.Gdk.Mobile
+{
+    public static class DeviceInfo
+    {
+        public const string AndroidEmulatorDefaultCallbackIp = "10.0.2.2";
```
Comment (zeroZshadow) [inline workers/unity/Packages/com.improbable.gdk.mobile/Utility/DeviceInfo.cs]: What is this ip?

File: workers/unity/Packages/com.improbable.gdk.mobile/Utility/LaunchArguments.cs [hunk 11]
```diff
@@ -14,9 +14,10 @@ public static class LaunchArguments
                 return new Dictionary<string, string>();
             }
 
+#if UNITY_ANDROID
             try
             {
-                using (var unityPlayer = new UnityEngine.AndroidJavaClass("com.unity3d.player.UnityPlayer"))
+                using (var unityPlayer = new AndroidJavaClass("com.unity3d.player.UnityPlayer"))
```
Comment (zeroZshadow) [inline workers/unity/Packages/com.improbable.gdk.mobile/Utility/LaunchArguments.cs]: If this is changed, should the 2 below also be cleaned up?

File: workers/unity/Packages/com.improbable.gdk.mobile/Worker/DefaultMobileWorkerConnector.cs [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+                WorkerType = workerType,
+                Network =
+                {
+                    ConnectionType = NetworkConnectionType.Kcp,
+                    UseExternalIp = true,
+                    Kcp = new KcpNetworkParameters
+                    {
+                        Heartbeat = new HeartbeatParameters()
+                        {
+                            IntervalMillis = 5000,
```
Comment (zeroZshadow) [inline workers/unity/Packages/com.improbable.gdk.mobile/Worker/DefaultMobileWorkerConnector.cs]: These 2 values might need a comment as to why they are those values

Comment (jessicafalk) [reply workers/unity/Packages/com.improbable.gdk.mobile/Worker/DefaultMobileWorkerConnector.cs]: these values might change. Need to double-check with people, so would prefer not to do this now. The values have never changed since we introduced them btw

File: workers/unity/Packages/com.improbable.gdk.mobile/Utility/DeviceInfo.cs (hunk 10 above)
Comment (jessicafalk) [reply workers/unity/Packages/com.improbable.gdk.mobile/Utility/DeviceInfo.cs]: the ip necessary to connect your emulator to your localhost
